packages/editor-ui/src/utils.ts

```diff
@@ -62,3 +63,65 @@ export const intersection = <T>(...arrays: T[][]): T[] => {
 	const ab = a.filter(v => b.includes(v));
 	return [...new Set(rest.length ? intersection(ab, ...rest) : ab)];
 };
+
+export const isObj = (obj: unknown): obj is object => !!obj && Object.getPrototypeOf(obj) === Object.prototype;
+
+export const isSchemaTypeObjectOrList = (type: string) => ['object', 'list'].includes(type);
+
+export const getTypeof = (value: unknown): N8nJsonSchemaType => value === null
+	? 'null'
+	: value instanceof Date
+		? 'date'
+		: Array.isArray(value)
+			? 'list'
+			: typeof value;
+
+export const getJsonSchema = (input: Optional<Primitives | object>, key?: string, path = ''): N8nJsonSchema => {
+	let schema:N8nJsonSchema = { type: 'undefined', value: 'undefined' };
+	switch (typeof input) {
+		case 'object':
+			if (input === null) {
+				schema = { type: 'string', value: '[null]' };
+			} else if (input instanceof Date) {
+				schema = { type: 'date', value: input.toISOString() };
+			} else if (Array.isArray(input)) {
+				schema = {
+					type: 'list',
+					value: '',
+					path: `${path}[*]`,
+				};
+				const firstItem = input[0];
+				if(Array.isArray(firstItem)){
+					schema.value = [getJsonSchema(firstItem, '', `${path}[*]`)];
+				} else if(isObj(firstItem)){
+					schema.value = getJsonSchema(firstItem, '', `${path}[*]`).value;
+				} else {
+					schema.value = getTypeof(firstItem);
+				}
+			} else if (isObj(input)) {
+				schema = {
+					type: 'object',
+					value: Object.entries(input).map(([k, v]) => ({ key: k, ...getJsonSchema(v, k, path + `.${k}`)})),
```

Comment:
what if the key has spaces or dots in it? then this path would not work... for example a json object like `{"hello world": {"test.": 1}}`
I would recommend defaulting to `x["hello world"]["test."]` syntax